astropy/io/ascii/qdp.py

```diff
@@ -190,7 +190,7 @@ def _interpret_err_lines(err_specs, ncols, names=None):
         ...
     ValueError: Inconsistent number of input colnames
     """
-    colnames = ["" for i in range(ncols)]
+    colnames = [None] * ncols
```

Comment:
What's happening with these changes? I don't know this code but these changes seem more substantial than I would expect.

Reply:
Previously assert was doing double duty as a sanity check for the interpretation code that shouldn't normally be seen with standard datatypes and also a check for empty strings in `names`.
```py
>>> err_specs = {'terr': [1], 'serr': [2]}
>>> ncols = 5
>>> col_in = ['MJD', '']
>>> _interpret_err_lines(err_specs, ncols, names=col_in)
...
AssertionError
```
 I split out the checking for empty strings part to be earlier and throw a descriptive `ValueError`. I think it makes more sense for the failure placeholder to be `None` instead of `""`.

since _I think_ it's just a sanity check at the end, I guess an an `AssertionError` or a plain `RuntimeError` would make more sense at the end

@matteobachetti am I interpreting the logic correctly?

Reply:
I'd say that... if what comes out of the function is exactly the same as it was before, the new logic is as good as the old one. Maybe, given the addition at line 205, we can make a test feeding empty strings as the name of a couple of columns and see what happens?

Reply:
@matteobachetti - if you review the changes and sign off then that sounds great!

Reply:
@taldcroft I need the tests to run, to verify that there are no side effects. I also suggested to add a test for the added logic. It might just be an additional line in the doctest verifying that the ValueError is indeed raised when feeding empty strings as names.

Reply:
> to verify that there are no side effects.

Indeed, this is why I am wary of a real logic change in a PR that should ideally be a no-op to code logic, especially in a bit of code that is somewhat complex.

I would strongly suggest in this PR to do the minimal change, which is replacing the one line:
```
assert not np.any([c == "" for c in colnames])
```
with
```
if np.any([c == "" for c in colnames]):
    raise ValueError("One or more columns could not be interpreted")
```
That will accomplish the stated goal of this PR with no logic change to the QDP parser. If scrutiny of the QDP code reveals it should be improved then that should be done in a separate PR.